simln-lib/src/lib.rs

```diff
@@ -620,10 +621,42 @@ struct ExecutorKit {
     source_info: NodeInfo,
     /// We use an arc mutex here because some implementations of the trait will be very expensive to clone.
     /// See [NetworkGraphView] for details.
-    network_generator: Arc<Mutex<dyn DestinationGenerator>>,
+    network_generator: Arc<dyn DestinationGenerator>,
     payment_generator: Box<dyn PaymentGenerator>,
 }
 
+struct PaymentEvent {
+    pubkey: PublicKey,
+    absolute_time: SystemTime,
+    destination: NodeInfo,
+    capacity: Option<u64>,
+}
+
+struct PaymentEventPayload {
+    executor: ExecutorKit,
+    current_count: u64,
```

Comment:
Thoughts on making this an `Option`? Since we don't need it for payments that aren't count-bounded.

Reply:
I think it is not necessary. It will make the code unnecessarily more verbose. Right now, the counter is only used here. 
```rust
if let Some(c) = payment_tracker.executor.payment_generator.payment_count() {
    if c == payment_tracker.payments_completed {
        log::info!( "Payment count has been met for {}: {c} payments. Stopping the activity.", payment_tracker.executor.source_info);
        continue
    }
}
```
And also in `get_payment_delay`, which does not accept an  `option`